### ISSUE TITLE
orderBy using eagerAlgorithm

### ISSUE THREAD

**Issue by matias025** [Vincit/objection.js#345]
Hi! After all, sorry about my english, I speak spanish and I'm using google translate...
I'm traying to use eagerAlgorithm, modify and orderBy in the same query, but the orderBy does not work.

Here is my code:
```js
function jornadaController () {

	this.getJornadas = function (params) {

		var filterResults = function(queryBuilder) {
			if (params != undefined) {
				if (params.agencia_id != ''){
					if (params.terminal_id == ''){
						queryBuilder.where('terminal.agencia_id', params.agencia_id);
					} else {
						queryBuilder.where('jornadas.terminal_id', params.terminal_id);
					}
				}
				if (params.fecha_desde != '') {
					queryBuilder.where('jornadas.fecha', '>=', params.fecha_desde);
				}
				if (params.fecha_desde != '' && params.fecha_hasta != '') {
					queryBuilder.where('jornadas.fecha', '<=', params.fecha_hasta);
				}
			}
		}

		return Jornada
			.query()
			.eagerAlgorithm(Jornada.JoinEagerAlgorithm)
			.eager('terminal.agencia')
			.modify(filterResults)
			.orderBy('jornadas.fecha', 'DESC')
			.debug(true)
			.then(jornadas => {
				return jornadas;
			})
			.catch(() => {
				return false;
			});
	}
}
```
However, when I run the query that returns .debug () in pgadmin, it works.

I do not know if I'm doing something wrong, or it's a framework issue

Thanks! Matias

**Comment by koskimas** [comment]
Could you be more specific: what happens, and what do you expect to happen? What's the query printed out by `.debug()`.

**Comment by matias025** [comment]
Thanks for the reply!
I pretend that the result is sorted by the field "fecha", but that does not happen, the result is sorted by the field "id".

The printed out query by .debug() is:

`
select "jornadas"."id" as "id", "jornadas"."terminal_id" as "terminal_id", "jornadas"."cantidad_tickets" as "cantidad_tickets", "jornadas"."total_apostado" as "total_apostado", "jornadas"."total_pagado" as "total_pagado", "jornadas"."total_tickets" as "total_tickets", "jornadas"."fecha_ultimo_registro" as "fecha_ultimo_registro", "jornadas"."total_cargas_manuales" as "total_cargas_manuales", "jornadas"."total_retenciones" as "total_retenciones", "jornadas"."fecha" as "fecha", "jornadas"."hora" as "hora", "jornadas"."total_cargas_billetes" as "total_cargas_billetes", "jornadas"."total_tickets_pagados" as "total_tickets_pagados", "terminal"."id" as "terminal:id", "terminal"."numero" as "terminal:numero", "terminal"."agencia_id" as "terminal:agencia_id", "terminal"."activa" as "terminal:activa", "terminal"."credito_disponible" as "terminal:credito_disponible", "terminal:agencia"."id" as "terminal:agencia:id", "terminal:agencia"."numero" as "terminal:agencia:numero", "terminal:agencia"."domicilio" as "terminal:agencia:domicilio", "terminal:agencia"."telefono" as "terminal:agencia:telefono", "terminal:agencia"."limite_renta" as "terminal:agencia:limite_renta", "terminal:agencia"."propietario" as "terminal:agencia:propietario", "terminal:agencia"."version" as "terminal:agencia:version", "terminal:agencia"."ciudad_id" as "terminal:agencia:ciudad_id", "terminal:agencia"."activa" as "terminal:agencia:activa", "terminal:agencia"."creacion" as "terminal:agencia:creacion", "terminal:agencia"."nombre" as "terminal:agencia:nombre" from "jornadas" as "jornadas" left join "terminales" as "terminal" on "terminal"."id" = "jornadas"."terminal_id" left join "agencias" as "terminal:agencia" on "terminal:agencia"."id" = "terminal"."agencia_id" order by "jornadas"."fecha" DESC
`

Thanks again!!

**Comment by koskimas** [comment]
This is probably a bug in objection. Thanks for reporting this! I'll make sure that's the case and fix this on monday.

**Comment by matias025** [comment]
Thank to you!

**Comment by koskimas** [comment]
This bug is fixed in objection 0.7.12.

**Comment by matias025** [comment]
It works fine! thank you very much
